ppa/archive/tests/test_models.py

```diff
@@ -15,13 +15,51 @@
 import pytest
 
 from ppa.archive import hathi
-from ppa.archive.models import DigitizedWork, Collection, NO_COLLECTION_LABEL
+from ppa.archive.models import DigitizedWork, Collection, \
+    NO_COLLECTION_LABEL, ProtectedFlags
 from ppa.archive.solr import get_solr_connection, Indexable
 
 
 FIXTURES_PATH = os.path.join(settings.BASE_DIR, 'ppa', 'archive', 'fixtures')
 
 
+class TestProtectedFlags(TestCase):
+
+    def test_deconstruct(self):
+        ret = ProtectedFlags.deconstruct()
+        assert ret[0] == 'ppa.archive.models.ProtectedFlags'
+        assert ret[1] == ['no_flags']
+        assert ret[2] == {}
+
+    def test_all_fields(self):
+        fields = ProtectedFlags.all_fields()
+        assert isinstance(fields, list)
+        # Trivial test, but will catch changes to the return signature
+        # that aren't a simple listing of members
+        assert fields == list(ProtectedFlags.__members__.keys())
+
+    def test_as_list(self):
+        # should return a list of the flags that have been set
+        results = ProtectedFlags.title | ProtectedFlags.subtitle
+        assert sorted(results.as_list()) == ['subtitle', 'title']
+
+    def test_str(self):
+        fields = ProtectedFlags.enumcron | ProtectedFlags.title | \
+            ProtectedFlags.sort_title
+
+        comparison = [
+            DigitizedWork._meta.get_field('enumcron').verbose_name,
+            DigitizedWork._meta.get_field('sort_title').verbose_name.capitalize(),
+            # no verbose name, so should be capitalized
+            DigitizedWork._meta.get_field('title').verbose_name.capitalize()
+        ]
+        # should be a comma-joined list of the verbose names of the
+        # fields, in alphabetical order; avoided hard coding in case
+        # verbose name changes
+        assert str(fields) == ', '.join(comparison)
```

Comment:
I find this test hard to read; I wonder if the verbose name logic should be split out so it's testable directly?

Reply:
Rendered a non-issue by not needing the verbose fields